server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -363,7 +370,7 @@ private static void nested(DocumentParserContext context, NestedObjectMapper nes
     private static void addFields(Version indexCreatedVersion, LuceneDocument nestedDoc, LuceneDocument rootDoc) {
         String nestedPathFieldName = NestedPathFieldMapper.name(indexCreatedVersion);
         for (IndexableField field : nestedDoc.getFields()) {
-            if (field.name().equals(nestedPathFieldName) == false) {
+            if (field.name().equals(nestedPathFieldName) == false && field.name().equals(IdFieldMapper.NAME) == false) {
```

Comment:
Should we move the logic for adding _id fields to nested docs to the `IdFieldMapper.postParse` method, similar as we have it for other metadata fields? Then we wouldn't need this exception here?

Reply:
That sounds like a good plan, but I think it might end up being a fairly big change as there are a few places that refer directly to the id field scattered around the place.  It also interacts with TSDB in a slightly complicated way - are you OK with doing this as a follow-up?

Reply:
sure, follow-up is fine.